src/main/resources/dependency-plugin.js

```diff
@@ -1,17 +1,22 @@
 const writeStats = (compilation) => {
-  const ms = [];
-  for (let module of compilation.getStats().toJson().modules) {
-    let reasons = [];
-    for (let reason of module.reasons) {
-      reasons.push(reason.moduleName);
+  const tree = [];
+
+  for (const chunk of compilation.chunks) {
+    const obj = {};
+    const deps = [];
+
+    for (const m of chunk._modules) {
+      deps.push(m.id);
     }
-    ms.push({
-      name: module.name,
-      reasons: reasons
-    })
+
+    obj.name = chunk.files[0]; // After much research, I have yet figure out WHY this is in an array, but it is, and at least in our case it is always an array of 1
```

Comment:
It's probably just the way it is. Your comment to clarify it is good.

I'd like this to be simplified even more. We can do:

```
tree.push({
  name: chunk.files[0], // After much research, I have yet figure out WHY this is in an array, but it is, and at least in our case it is always an array of 1
  dependencies: deps
})
```

Reply:
I will take care of both of those today.